exercises/rail-fence-cipher/rail_fence_cipher_test.rb

```diff
@@ -0,0 +1,94 @@
+gem 'minitest', '>= 5.0.0'
+require 'minitest/autorun'
+require_relative 'rail_fence_cipher'
+
+# rubocop:enable all
+class RailFenceCipherTest < Minitest::Test
+  def test_encode_with_empty_string
+    assert_equal '', RailFenceCipher.encode('', 4)
+  end
+
+  def test_encode_with_one_rail
+    skip
+    assert_equal 'One rail, only one rail',
+                 RailFenceCipher.encode('One rail, only one rail', 1)
+  end
+
+  def test_encode_with_two_rails
+    skip
+    assert_equal 'XXXXXXXXXOOOOOOOOO',
+                 RailFenceCipher.encode('XOXOXOXOXOXOXOXOXO', 2)
+  end
+
+  def test_encode_with_three_rails
+    skip
+    assert_equal 'WECRLTEERDSOEEFEAOCAIVDEN',
+                 RailFenceCipher.encode('WEAREDISCOVEREDFLEEATONCE', 3)
+  end
+
+  def test_encode_with_ending_in_the_middle
+    skip
+    assert_equal 'ESXIEECSR', RailFenceCipher.encode('EXERCISES', 4)
+  end
+
+  def test_encode_with_a_more_diverse_text
+    skip
+    input = 'The quick brown fox jumps over the lazy dog.'
+    assert_equal 'Tioxs aghucrwo p rtlzo.eqkbnfjmoeh yd   uve ',
+                 RailFenceCipher.encode(input, 4)
+  end
+
+  def test_encode_with_less_letters_than_rails
+    skip
+    assert_equal 'More rails than letters',
+                 RailFenceCipher.encode('More rails than letters', 24)
+  end
+
+  def test_decode_with_empty_string
+    skip
+    assert_equal '', RailFenceCipher.decode('', 4)
+  end
+
+  def test_decode_with_one_rail
+    skip
+    assert_equal 'ABCDEFGHIJKLMNOP',
+                 RailFenceCipher.decode('ABCDEFGHIJKLMNOP', 1)
+  end
+
+  def test_decode_with_two_rails
+    skip
+    assert_equal 'XOXOXOXOXOXOXOXOXO',
+                 RailFenceCipher.decode('XXXXXXXXXOOOOOOOOO', 2)
+  end
+
+  def test_decode_with_three_rails
+    skip
+    assert_equal 'THEDEVILISINTHEDETAILS',
+                 RailFenceCipher.decode('TEITELHDVLSNHDTISEIIEA', 3)
+  end
+
+  def test_decode_with_a_more_diverse_text
```

Comment:
Yeah, tests are awesome during development, but you have found that the same logic applies to all cases now, right? Maybe this could even keep people wondering if they have to implement something different for texts with 4 or more fences, for example.